packages/embark/src/lib/modules/coverage/index.ts

```diff
@@ -54,9 +49,9 @@ export default class Coverage {
   }
 
   private swapContracts() {
-    this.embark.config.embarkConfig.contracts = this.contractsDir.reduce((acc: string[], value: string) => (
-      acc.concat(path.join(coverageContractsPath(), value))
-    ), []);
+    this.embark.config.embarkConfig.contracts = this.originalContractFiles.map((file: File) => (
+      path.join(coverageContractsPath(), file.path)
+    ));
```

Comment:
Didn't realize we modified the configs from here. Is it safe to do so? Should we use an event instead?

Reply:
Yes it is safe, but is it the best we can do?
This is about knowledge, coverage know about config but config doesn't know about coverage in this scenario.
I always pictured config as transient (part of core, so available everywhere)
To rephrase it: Core is available as an API to all module so I never expect event in core.
I can make the changes if you think I am wrong

Reply:
Ok, makes sense. 
On another note, doesn't this line change the `contracts` array to only contain paths when it should be containing files?